devtools/conda-envs/test_env.yaml

```diff
@@ -5,7 +5,7 @@ channels:
 dependencies:
   - python
   - pip
-  - numpy==1.18.0
+  - numpy
```

Comment:
no need to pin to a version here, we can assume that the highest version will be picked up (with the minimum tests you then have reasonable confidence that the whole range of numpy versions will work).

Reply:
Ok, I think my understanding of this was quite the opposite. I am missing a key point here. The premise I had was "I am going to test the minimum version of Numpy for each version of Python. Then I can identify what's my bottom end, and therefore, I identify what is the min version supported in my repo". 
You say 
> we can assume that the highest version will be picked up

Why would I be interested in the highest version? Shouldn't I be targeting the min version for each version of Python?

Reply:
My response https://github.com/MDAnalysis/membrane-curvature/pull/51#discussion_r674408301 above probably answers this to an extent.

The short answer is that as long as you test the two extremes you can be somewhat confident that the intermediate versions will behave properly, especially if you eventually follow NEP29 and don't have too large a gap between numpy versions.

Reply:
Thanks, I was about to reply along the same lines. So, to finish this topic, would you please confirm the following statement is correct? "The min version I am going to put on my setup.py is the min version I test, and I will assume that all in between that lowest end and the highest will also work." (I get the highest version from the `test_env.yaml` when I create the environment)

Reply:
Yeah, without pinning `test_env.yaml` should pick up the latest version assuming it's compatible with all your dependencies.

Technically if you wanted to be super accurate / strict you could also put a maximum version in your setup.py, but just putting the lowest version you've tested on is fine as long as you've tested the most recent version at the time of release.